src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -300,22 +301,24 @@ private void HandleCompletionRequest(string id, string paramsJson)
             // Avoid this if possible
             _logger?.Invoke($"[PFX] HandleCompletionRequest: Suggest results: Count:{result.Suggestions.Count()}, Suggestions:{string.Join(", ", result.Suggestions.Select(s => $@"[{s.Kind}]: '{s.DisplayText.Text}'"))}");
 
-            var items = new List<CompletionItem>();
-
-            foreach (var item in result.Suggestions)
+            var precedingCharacter = expression[cursorPosition - 1];
+            _sendToClient(JsonRpcHelper.CreateSuccessResult(id, new
             {
-                items.Add(new CompletionItem()
+                items = result.Suggestions.Select((item, index) => new CompletionItem()
                 {
                     Label = item.DisplayText.Text,
                     Detail = item.FunctionParameterDescription,
                     Documentation = item.Definition,
-                    Kind = GetCompletionItemKind(item.Kind)
-                });
-            }
+                    Kind = GetCompletionItemKind(item.Kind),
 
-            _sendToClient(JsonRpcHelper.CreateSuccessResult(id, new
-            {
-                items,
+                    // The order of the results should be preserved.  To do that, we embed the index
+                    // into the sort text, which clients may sort lexigraphically.
+                    SortText = index.ToString(CultureInfo.InvariantCulture),
+
+                    // If the current position is in front of a single quote and the completion result starts with a single quote,
+                    // we don't want to make it harder on the end user by inserting an extra single quote.
+                    InsertText = item.DisplayText.Text is { } label && label[0] == '\'' && precedingCharacter == '\'' ? label.Substring(1) : item.DisplayText.Text
```

Comment:
Could we use the TexlLexer.IsIdentDelimiter(label[0]) instead of hard coding the delimiter? also helps if fx changes ident delimiter in future (less likely as it is GAed now)

Reply:
Yes this is a good idea